<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1000</width>
    <height>800</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>0</width>
    <height>800</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>1000</width>
    <height>800</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Sign Language Translator</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>40</number>
    </property>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <property name="topMargin">
       <number>20</number>
      </property>
      <property name="bottomMargin">
       <number>20</number>
      </property>
      <item alignment="Qt::AlignHCenter">
       <widget class="QLabel" name="titleLabel">
        <property name="font">
         <font>
          <family>Century Gothic</family>
          <pointsize>18</pointsize>
         </font>
        </property>
        <property name="text">
         <string>Sign Language Translator</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <property name="spacing">
         <number>40</number>
        </property>
        <property name="leftMargin">
         <number>100</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>100</number>
        </property>
        <item>
         <widget class="QLabel" name="videoLabel">
          <property name="minimumSize">
           <size>
            <width>400</width>
            <height>300</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>400</width>
            <height>300</height>
           </size>
          </property>
          <property name="styleSheet">
           <string notr="true">QLabel {
    border: 2px solid transparent;                          
    color: black;          
    transition: background-color 0.3s ease; 
}</string>
          </property>
          <property name="frameShape">
           <enum>QFrame::NoFrame</enum>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="scaledContents">
           <bool>true</bool>
          </property>
          <property name="indent">
           <number>0</number>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <property name="spacing">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_3">
            <property name="spacing">
             <number>0</number>
            </property>
            <item alignment="Qt::AlignHCenter">
             <widget class="QPushButton" name="prevButton">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>0</width>
                <height>240</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>40</width>
                <height>240</height>
               </size>
              </property>
              <property name="font">
               <font>
                <pointsize>16</pointsize>
               </font>
              </property>
              <property name="styleSheet">
               <string notr="true">QPushButton {
    border: 2px solid transparent;
    border-top-left-radius: 15px;      
	border-bottom-left-radius: 15px;    
    padding: 5px;                  
    background-color: white;        
    color: black;             
    transition: background-color 0.3s ease; 
}

QPushButton:hover {
    background-color: #f0f0f0; 
    color: black;                  
}

QPushButton:pressed {
    background-color: #d0d0d0;
    color: black;
}</string>
              </property>
              <property name="text">
               <string>🡸</string>
              </property>
             </widget>
            </item>
            <item alignment="Qt::AlignHCenter">
             <widget class="QLabel" name="slideshowLabel">
              <property name="minimumSize">
               <size>
                <width>240</width>
                <height>240</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>240</width>
                <height>240</height>
               </size>
              </property>
              <property name="styleSheet">
               <string notr="true">QLabel {
    border: 2px solid transparent;                            
    color: black;           
    transition: background-color 0.3s ease; 
}</string>
              </property>
              <property name="text">
               <string/>
              </property>
              <property name="scaledContents">
               <bool>false</bool>
              </property>
             </widget>
            </item>
            <item alignment="Qt::AlignHCenter">
             <widget class="QPushButton" name="nextButton">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="maximumSize">
               <size>
                <width>40</width>
                <height>240</height>
               </size>
              </property>
              <property name="font">
               <font>
                <pointsize>14</pointsize>
               </font>
              </property>
              <property name="styleSheet">
               <string notr="true">QPushButton {
    border: 2px solid transparent;
    border-top-right-radius: 15px;      
	border-bottom-right-radius: 15px;    
    padding: 5px;                  
    background-color: white;        
    color: black;             
    transition: background-color 0.3s ease; 
}

QPushButton:hover {
    background-color: #f0f0f0; 
    color: black;                  
}

QPushButton:pressed {
    background-color: #d0d0d0;
    color: black;
}</string>
              </property>
              <property name="text">
               <string>🡺</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item alignment="Qt::AlignHCenter">
           <widget class="QLabel" name="signLabel">
            <property name="minimumSize">
             <size>
              <width>150</width>
              <height>0</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>150</width>
              <height>40</height>
             </size>
            </property>
            <property name="font">
             <font>
              <family>Century Gothic</family>
              <pointsize>16</pointsize>
             </font>
            </property>
            <property name="styleSheet">
             <string notr="true">QLabel {
    border: 2px solid transparent;
    border-radius: 15px;      
    padding: 5px;                  
    background-color: rgb(214, 214, 214);        
    color: black;             
    transition: background-color 0.3s ease; 
}</string>
            </property>
            <property name="text">
             <string>TextLabel</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item alignment="Qt::AlignHCenter">
       <widget class="QLabel" name="predictionLabel">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>500</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>60</height>
         </size>
        </property>
        <property name="font">
         <font>
          <family>Century Gothic</family>
          <pointsize>20</pointsize>
         </font>
        </property>
        <property name="styleSheet">
         <string notr="true">QLabel {
    border: 2px solid transparent;
    border-radius: 15px;      
    padding: 5px;                  
    background-color: rgb(214, 214, 214);        
    color: black;             
    transition: background-color 0.3s ease; 
}</string>
        </property>
        <property name="text">
         <string>TextLabel</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="buttonsLayout" stretch="0,0,0,0,0,0">
        <property name="spacing">
         <number>20</number>
        </property>
        <property name="leftMargin">
         <number>50</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>50</number>
        </property>
        <item>
         <widget class="QPushButton" name="autoButton">
          <property name="font">
           <font>
            <family>Century Gothic</family>
            <pointsize>12</pointsize>
           </font>
          </property>
          <property name="styleSheet">
           <string notr="true">QPushButton {
    border: 2px solid transparent;
    border-radius: 15px;      
    padding: 5px;                  
    background-color: white;        
    color: black;             
    transition: background-color 0.3s ease; 
}

QPushButton:hover {
    background-color: #f0f0f0; 
    color: black;                  
}

QPushButton:pressed {
    background-color: #d0d0d0;
    color: black;
}</string>
          </property>
          <property name="text">
           <string>⚙️ Auto</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="caseButton">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="font">
           <font>
            <family>Century Gothic</family>
            <pointsize>12</pointsize>
           </font>
          </property>
          <property name="styleSheet">
           <string notr="true">QPushButton {
    border: 2px solid transparent;
    border-radius: 15px;      
    padding: 5px;                  
    background-color: white;        
    color: black;             
    transition: background-color 0.3s ease; 
}

QPushButton:hover {
    background-color: #f0f0f0; 
    color: black;                  
}

QPushButton:pressed {
    background-color: #d0d0d0;
    color: black;
}</string>
          </property>
          <property name="text">
           <string>Aa</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="addButton">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="font">
           <font>
            <family>Century Gothic</family>
            <pointsize>12</pointsize>
           </font>
          </property>
          <property name="styleSheet">
           <string notr="true">QPushButton {
    border: 2px solid transparent;
    border-radius: 15px;      
    padding: 5px;                  
    background-color: white;        
    color: black;             
    transition: background-color 0.3s ease; 
}

QPushButton:hover {
    background-color: #f0f0f0; 
    color: black;                  
}

QPushButton:pressed {
    background-color: #d0d0d0;
    color: black;
}</string>
          </property>
          <property name="text">
           <string>➕  Add</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="deleteButton">
          <property name="font">
           <font>
            <family>Century Gothic</family>
            <pointsize>12</pointsize>
           </font>
          </property>
          <property name="styleSheet">
           <string notr="true">QPushButton {
    border: 2px solid transparent;
    border-radius: 15px;      
    padding: 5px;                  
    background-color: white;        
    color: black;             
    transition: background-color 0.3s ease; 
}

QPushButton:hover {
    background-color: #f0f0f0; 
    color: black;                  
}

QPushButton:pressed {
    background-color: #d0d0d0;
    color: black;
}</string>
          </property>
          <property name="text">
           <string>🗑️ Delete</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="spaceButton">
          <property name="font">
           <font>
            <family>Century Gothic</family>
            <pointsize>12</pointsize>
           </font>
          </property>
          <property name="styleSheet">
           <string notr="true">QPushButton {
    border: 2px solid transparent;
    border-radius: 15px;      
    padding: 5px;                  
    background-color: white;        
    color: black;             
    transition: background-color 0.3s ease; 
}

QPushButton:hover {
    background-color: #f0f0f0; 
    color: black;                  
}

QPushButton:pressed {
    background-color: #d0d0d0;
    color: black;
}</string>
          </property>
          <property name="text">
           <string>Space</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="resetButton">
          <property name="font">
           <font>
            <family>Century Gothic</family>
            <pointsize>12</pointsize>
           </font>
          </property>
          <property name="styleSheet">
           <string notr="true">QPushButton {
    border: 2px solid rgb(203, 14, 70);
    border-radius: 15px;      
    padding: 5px;                  
    background-color: white;        
    color: black;             
    transition: background-color 0.3s ease; 
}

QPushButton:hover {
    background-color: #f0f0f0; 
    color: black;                  
}

QPushButton:pressed {
    background-color: #d0d0d0;
    color: black;
}</string>
          </property>
          <property name="text">
           <string>🗘 Reset</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item alignment="Qt::AlignHCenter">
       <widget class="QTextEdit" name="textBox">
        <property name="minimumSize">
         <size>
          <width>700</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>1000</width>
          <height>200</height>
         </size>
        </property>
        <property name="font">
         <font>
          <family>Century Gothic</family>
          <pointsize>16</pointsize>
         </font>
        </property>
        <property name="styleSheet">
         <string notr="true">QTextEdit {
    border: 2px solid transparent;
    border-radius: 15px;      
    padding: 15px;                  
    background-color: white;        
    color: black;             
    transition: background-color 0.3s ease; 
}</string>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Sunken</enum>
        </property>
        <property name="lineWidth">
         <number>6</number>
        </property>
        <property name="readOnly">
         <bool>false</bool>
        </property>
        <property name="placeholderText">
         <string>Waiting for input...</string>
        </property>
       </widget>
      </item>
      <item alignment="Qt::AlignHCenter">
       <widget class="QPushButton" name="saveToTextButton">
        <property name="minimumSize">
         <size>
          <width>200</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>200</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="font">
         <font>
          <family>Century Gothic</family>
          <pointsize>12</pointsize>
         </font>
        </property>
        <property name="styleSheet">
         <string notr="true">QPushButton {
    border: 2px solid transparent;
    border-radius: 15px;      
    padding: 5px;                  
    background-color: white;        
    color: black;             
    transition: background-color 0.3s ease; 
}

QPushButton:hover {
    background-color: #f0f0f0; 
    color: black;                  
}

QPushButton:pressed {
    background-color: #d0d0d0;
    color: black;
}</string>
        </property>
        <property name="text">
         <string>Save to Text</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
